security/auth/maskinporten.md

```diff
@@ -0,0 +1,141 @@
+---
+description: >
+  Enabling `server-2-server` authentication with external agencies using Maskinporten.
+---
+
+# Maskinporten Client
+
+{% hint style="warning" %}
+**Status: Opt-In Open Beta**
+
+This feature is only available in the [Google Cloud Platform (GCP)](../../clusters/gcp.md) clusters.
+{% endhint %}
+
+## Abstract
+
+Maskinporten is a service that offers a simple API security model based on the OAuth2 protocol, and the use of JWT bearer grants. A Concept inspired by [Google's System Accounts](https://developers.google.com/identity/protocols/oauth2/service-account).
+
+Maskinporten allows API providers (external agencies) to define access to their APIs, modeled as scopes and based on the consumer's organization number.
+
+The NAIS platform provides support for simple declarative provisioning of an [Maskinporten client](https://difi.github.io/felleslosninger/maskinporten_auth_server-to-server-oauth2.html) with sensible defaults that your application may use to integrate with Maskinporten.
+
+An Maskinporten `client` allows your application to leverage Maskinporten for authentication and authorization requesting external agencies. To achieve this, your application must implement [JWT grants](https://difi.github.io/felleslosninger/maskinporten_protocol_token.html).
+
+## Configuration
+
+### Getting Started
+
+{% code-tabs-item title="Minimal nais.yaml example" %}
+```yaml
+apiVersion: "nais.io/v1alpha1"
+kind: "Application"
+metadata:
+  name: nais-testapp
+  namespace: aura
+  labels:
+    team: aura
+spec:
+  image: navikt/nais-testapp:66.0.0
+  maskinporten:
+    enabled: true
+    scopes:
+      - scope: "nav:some/scope"
+```
+{% endcode-tabs-item %}
+
+### Spec
+
+See the [NAIS manifest](../../nais-application/reference.md#spec-maskinporten).
+
+### Access Policies
+
+The following [outbound external hosts](../../nais-application/access-policy.md#external-services) are automatically added when enabling this feature:
+
+- `ver2.maskinporten.no` in development
+- `maskinporten.no` in production
+
+You do not need to specify these explicitly.
+
+### Scopes
+
+Maskinporten allows API providers to define access to their APIs, modeled as scopes and based on the consumer's organization number.
+
+When a `client` requests Maskinporten (Identity Provider) for a token:
+- Maskinporten validates the validity of the JWT and signature ([Runtime JWK Secret](#runtime-variables-&-credentials) used to sign the JWT).  
+- When `client` has access to the requested resources: `scope`, an `access_token` will be returned to the `client` and can be used for further actions.
+
+For more details see: [Overview of flows in Digdirator](#overview-of-flows)
+
+{% hint style="warning" %}
+Make sure **NAV** have `pre-registered` rights to **all** `scopes` specified in the manifest or provision o client will fail.
```

Comment:
```suggestion
Make sure that the relevant service providers have pre-registered **NAV** as a valid consumer of any scopes that you define. Provisioning of client will fail otherwise.
```